runtime/str.go

```diff
@@ -717,6 +757,23 @@ func adjustIndex(i, l int) int {
 	return i
 }
 
+func adjustBounds(f *Frame, o *Object, l int, upperBound bool) (int, *BaseException) {
```

Comment:
Thanks for the implmentation. I added addictional check start > l for the case like `''.find('', 1, 1) `